hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/LongSampleAccumulator.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.mono.utils;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A statistics "counter" that collects (reduces) `Long` samples, keeping track of the number of
+ * samples, the total accumulated, and the maximum seen.
+ */
+public final class LongSampleAccumulator {
```

Comment:
The Platform metrics API already provides `com.swirlds.common.metrics.LongAccumulator`. That interface (+ implementation) should be used.

Reply:
Yes, except I want max as well and don't know how to get it from an accumulator (is there a way?)

Reply:
ok, but you now introduce a 3rd API. Let's wait on the comment from @netopyr

Reply:
well "being redone" probably was a paraphrase of _what you said_: that there's going to be a more cool stuff coming. let me look for it.

so ... it's an accepted thing in the devops/oncall world to combine metrics (such as in the way you describe) in the visualization tools? I'll consider that.  But really, what I'm trying to do here is a poor man's replacement for a histogram to sort of get exceptional (high) values which would otherwise be lost due to sampling.  (Plus I completely missed that the running average _does_ give you max, sorry).